### PR TITLE
Added activity functions to solid solution, updated example document

### PR DESCRIPTION
**Pull request by bobmyhill**
Quite often, we want to use the activity of a particular endmember in a solid solution, rather than the excess gibbs free energy. This commit adds two attributes to solidsolution, activities and activity_coefficients, which output the overall activity and the proportion of this value attributable to nonideality.

### CODE REVIEW

**Comment by sannecottaar** [comment]
Looks good to me, as far as I understand this. It is spitting out a runtime warning when temperature is 0 for the benchmark_solidsolution.py, so the output should be fixed to pass that test.

**Comment by bobmyhill** [comment]
Fixed!

**Comment by bobmyhill** [comment]
Ok, should be good to go...

File: burnman/solidsolution.py [hunk 1]
```diff
@@ -122,9 +121,30 @@ def set_state(self, pressure, temperature):
...
+        """
+        if self.temperature > 1.e-10:
+            return np.exp(self.excess_partial_gibbs/(constants.gas_constant*self.temperature))
+        else:
+            return None
 
+    @material_property
+    def activity_coefficients(self):
+        """
+        Returns a list of endmember activity coefficients (the non-ideal part of the activities
```
Comment (tjhei) [inline burnman/solidsolution.py]: closing parens?

File: burnman/solidsolution.py [hunk 2]
```diff
@@ -122,9 +121,30 @@ def set_state(self, pressure, temperature):
         Mineral.set_state(self, pressure, temperature)
         for i in range(self.n_endmembers):
             self.endmembers[i][0].set_state(pressure, temperature)
+            
 
+    @material_property
+    def activities(self):
+        """
+        Returns a list of endmember activities
```
Comment (tjhei) [inline burnman/solidsolution.py]: units?

**Comment by tjhei** [comment]
it would also be great to have a test that somehow covers numeric values of these functions.

**Comment by bobmyhill** [comment]
Ok, I've changed the implementation to be a bit more efficient (there's no np.exp(np.log()) to calculate activities now). I've also added some tests and added internal and helmholtz energies to the solid solution calculations.

**Comment by tjhei** [comment]
I still have no idea what these activities are, but I am not a mineral physicist. :-)

Looks good to me.

**Comment by bobmyhill** [comment]
I suspected that this was the reason that it wasn't merged yet :) There's a good introduction here:
http://serc.carleton.edu/research_education/equilibria/activitymodels.html

Don't worry, I'll add some details to the documentation in a separate PR. Activities are very useful.

**Comment by bobmyhill** [comment]
@sannecottaar, are you also happy?

**Comment by sannecottaar** [comment]
Yes. I'm just going to go ahead and merge this.

**Pull request merged**